core/metamodel/src/main/java/org/apache/isis/core/metamodel/facets/object/hidden/HiddenTypeFacetDerivedFromAuthorization.java

```diff
@@ -29,14 +29,16 @@
 
 import lombok.val;
 
-public class HiddenTypeFacetDerivedFromAuthorization extends FacetAbstract implements HiddenTypeFacet {
+public class HiddenTypeFacetDerivedFromAuthorization
+extends FacetAbstract
+implements HiddenTypeFacet {
 
     public static Class<? extends Facet> type() {
         return HiddenTypeFacet.class;
     }
 
     public HiddenTypeFacetDerivedFromAuthorization(final FacetHolder holder) {
-        super(type(), holder);
+        super(type(), holder, Precedence.DERIVED);
```

Comment:
Although this facet's class name has the word "derived" in it, I'm not sure it automatically means that it's precedence is DERIVED.
The semantics of DERIVED (nb: this is not documented anywhere) is for a facet that is inferred from other facets, but which could be overridden by something that is more specific.
That isn't the case here: the type should always be hidden if all of the members are.
It should probably be named "HiddenTypeFacetInferredFromNoAuthorisationToAnyMember" or something similar.

Reply:
Thanks Dan! With this commit I was just preserving the original behavior. But I guess bugs do surface in the process. I'll change it then.

Reply:
Sure.  Some thoughts (probably obvious) on the other precedences:
* FALLBACK is lowest priority if there are no other facets for that type, eg infer the NamedFacet from the method name if not specified explicitly using an annotation or layout
* the notion of "alwaysReplace" I can't remember, but I suspect it is very similar to fallback.  
* the notion of an underlying facet is where a higher precedence facet can replace another facet, but keep hold of the underlying facet.  In some cases the replacing facet might wrap but delegate to the underlying, eg to translate the name facet.  If the replacing facet doesn't delegate to the underlying, then there's no real need to retain that reference, but (I wouldn't be surprised) if FacetUtil.addFacet(...) always just chains them anyway regardless.

Reply:
Thanks, indeed all correct, I agree.